www/pfSense-pkg-vHosts/files/usr/local/www/vhosts_certs.php

```diff
@@ -0,0 +1,465 @@
+<?php
+/*******************************************************************************
+* vHosts package certfificate manager page.  Drops the current list of certs or
+* a form page for adding/updating certs.
+*
+* GET: No parameters:
+*   Drops current list of certs with options to add/change/delete. Delete option
+*   is available only if cert is not in use.
+*
+* GET: 'act' = 'add'
+*   Drops form for input of new certificate and key.
+*
+* GET: 'act' = 'chg', 'id' = n
+*   Drops form for update of certificate 'n' settings.
+*   Form action returns 'id' value.
+*
+* GET: 'act' = 'del', 'id' = n
+*   Removes certificate 'n' and drops list page.
+*
+* GET: 'act' = 'show', 'id' = n
+*   Drops certificate 'n' details for modal dialog.
+*
+* POST: 'id' = n (optional)
+*   Adds or updates certificate config and drops list page.
+* ------------------------------------------------------------------------------
+* Part of pfSense 2.3 and later (https://www.pfSense.org/).
+* Copyright (c) 2004-2016 Rubicon Communications, LLC (Netgate)
+*
+* Inspired by vhosts package originally written by Mark Crane.
+* Copyright (C) 2008 Mark J Crane
+* Copyright (C) 2015 ESF, LLC
+* Copyright (C) 2016 Softlife Consulting
+* All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+require_once("guiconfig.inc");
+require_once("util.inc");
+require("/usr/local/pkg/vhosts.inc");
+
+$pgtitle   = [ gettext("Services"), gettext("vHosts"), gettext("Certificates") ];
+$pgtabs =
+	[
+	[ gettext("Hosts"),        false, "vhosts.php" ],
+	[ gettext("Certificates"), true,  "vhosts_certs.php" ]
+	];
+
+/*******************************************************************************
+* Main Line
+*******************************************************************************/
+$a_vcerts = &$config['installedpackages']['vhosts']['cert'];
+$act      = $_GET['act'];
+$id       = $_GET['id'];
```

Comment:
See comment in previous file about testing this and unsetting if it's not the proper type